src/wp-includes/media.php

```diff
@@ -2077,9 +2079,12 @@ function wp_print_auto_sizes_contain_css_fix() {
 		return;
 	}
 
-	?>
-	<style>img:is([sizes="auto" i], [sizes^="auto," i]) { contain-intrinsic-size: 3000px 1500px }</style>
-	<?php
+	$handle = 'wp-img-auto-sizes-contain';
+	wp_register_style( $handle, false );
+	wp_add_inline_style( $handle, 'img:is([sizes="auto" i], [sizes^="auto," i]) { contain-intrinsic-size: 3000px 1500px }' );
+
+	// Make sure inline style is printed first.
```

Comment:
```suggestion
	// Make sure inline style is printed first since it was previously printed at wp_head priority 1 and this preserves the CSS cascade.
```